### PR TITLE
fix: make usageplan properties referable

### PR DESCRIPTION
**Pull request by ShreyaGangishetty**
*Issue #, if available:*
#1432 

*Description of changes:*
made usageplan resources a referable property to api

*Description of how you validated changes:*

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [x] Verify transformed template deploys and application functions as expected
- [ ] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/sam_resources.py [hunk 1]
```diff
@@ -764,6 +771,9 @@ class SamApi(SamResourceMacro):
         "Stage": ApiGatewayStage.resource_type,
         "Deployment": ApiGatewayDeployment.resource_type,
         "DomainName": ApiGatewayDomainName.resource_type,
+        "UsagePlan": ApiGatewayUsagePlan.resource_type,
+        "UsagePlanKey": ApiGatewayUsagePlanKey.resource_type,
```
Comment (keetonian) [inline samtranslator/model/sam_resources.py]: `Ref` does nothing for UsagePlanKeys, we shouldn't include it here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-usageplankey.html

Comment (ShreyaGangishetty) [reply samtranslator/model/sam_resources.py]: `Ref` for `ApiGatewayUsagePlanKey` gives the physical id which has the mapping `ApiKeyPhysicalId:UsagePlanPhysicalId`. I think it was a miss in the docs

Comment (keetonian) [reply samtranslator/model/sam_resources.py]: Sounds good, this looks good to me then

**Comment by ShreyaGangishetty** [comment]
> Please add docs on referable properties

I will submit another PR for docs on referable properties that will also have other referable properties documented for `AWS::Serverless::Function`, `AWS::Serverless::Api` and `AWS::Serverless::HttpApi`

**Pull request merged**